src/main/java/org/xstefank/api/TeamCityAPI.java

```diff
@@ -0,0 +1,81 @@
+package org.xstefank.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.xml.bind.DatatypeConverter;
+import org.jboss.logging.Logger;
+import org.xstefank.model.Utils;
+import org.xstefank.model.json.triggerbuild.BuildJson;
+
+public class TeamCityAPI implements ContinuousIntegrationAPI {
+
+    private static final Logger log = Logger.getLogger(TeamCityAPI.class);
+
+    private final String baseUrl;
+    private final String encryptedCredentials;
+    private final String buildId;
+
+    public TeamCityAPI() {
+        this.baseUrl = getBaseUrl(Utils.getTyrProperty(Utils.TEAMCITY_HOST_PROPERTY),
+                Integer.parseInt(Utils.getTyrProperty(Utils.TEAMCITY_PORT_PROPERTY)));
+
+        this.encryptedCredentials = encryptCredentials(Utils.getTyrProperty(Utils.TEAMCITY_USER_PROPERTY),
+                Utils.getTyrProperty(Utils.TEAMCITY_PASS_PROPERTY));
+
+        this.buildId = Utils.getTyrProperty(Utils.TEAMCITY_BUILD_CONFIG);
+    }
+
+    TeamCityAPI(String host, int port, String username, String password, String buildId) {
+        this.baseUrl = getBaseUrl(host, port);
+        this.encryptedCredentials = encryptCredentials(username, password);
+        this.buildId = buildId;
```

Comment:
Can you elaborate a little on this property? Does it mean that each instance of TeamCityAPI is associated with only one build?